InteractionBase/src/handlers/GenericHandler.cpp

```diff
@@ -153,13 +154,55 @@ void GenericHandler::showCommandPrompt(Visualization::Item* commandReceiver, QSt
 {
 	if (commandPrompt_ && commandPrompt_->commandReceiver() == commandReceiver)
 	{
-		commandPrompt_->showPrompt(initialCommandText);
+        commandPrompt_->showPrompt(initialCommandText);
+        this->showCommandMenu(commandReceiver);
 	}
 	else
 	{
-		removeCommandPrompt();
-		commandPrompt_ = new CommandPrompt(commandReceiver, initialCommandText);
-	}
+        removeCommandPrompt();
+        commandPrompt_ = new CommandPrompt(commandReceiver, initialCommandText);
+        this->showCommandMenu(commandReceiver);
+	}
+}
+
+void GenericHandler::showCommandMenu(Visualization::Item* commandReceiver)
+{
+    QList<AutoCompleteEntry*> entries;
+    for (auto* command : commands())
+    {
+        if (command->canBeUsedInMenu())
+        {
+            //We find the first parent where it can be interpreted
+            auto target = commandReceiver;
+            while (target && !command->canInterpret(commandReceiver, target, QStringList(command->name()),
+                                                    commandPrompt_->commandReceiverCursor()))
+                target = target->parent();
+            //If we have a target != null, then we can interpret the command on it
+            //We then create the command with its default arguments essentially when it is selected
+            if (target)
+            {
+                auto suggestions = command->suggest(commandReceiver, target, command->name(),
+                                                    commandPrompt_->commandReceiverCursor());
+                for (auto* suggestion : suggestions)
+                    entries.append(new AutoCompleteEntry(suggestion->text(),
+                                                         suggestion->description(),
+                                                         nullptr,
+                                                         [commandReceiver, this, suggestion](AutoCompleteEntry*)
```

Comment:
You should not need to capture this or suggestion. What you need is the command and the suggestion text. Always capture the minimal information that you need.